sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c

```diff
@@ -0,0 +1,487 @@
+  memset(mqtt_sub_topic_buf, 0xFF, _az_COUNTOF(mqtt_sub_topic_buf));
+  az_span mqtt_sub_topic = az_span_init(mqtt_sub_topic_buf, 0, _az_COUNTOF(mqtt_sub_topic_buf));
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span request_id = AZ_SPAN_NULL;
+  uint16_t status = 202;
+
+  assert_precondition_checked(
+    az_iot_hub_client_methods_response_publish_topic_get(&client, request_id, status, mqtt_sub_topic, &mqtt_sub_topic)
+  );
+}
+
+static void test_az_iot_hub_client_methods_received_topic_parse_NULL_client_fail(void** state)
+{
+  (void)state;
+
+  az_span received_topic = AZ_SPAN_FROM_STR("$iothub/methods/POST/TestMethod/?$rid=1");
+
+  az_iot_hub_client_method_request out_request;
+
+  assert_precondition_checked(
+    az_iot_hub_client_methods_received_topic_parse(NULL, received_topic, &out_request)
+  );
+}
+
+static void test_az_iot_hub_client_methods_received_topic_parse_EMPTY_received_topic_fail(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span received_topic = AZ_SPAN_FROM_STR("");
+
+  az_iot_hub_client_method_request out_request;
+
+  assert_precondition_checked(
+    az_iot_hub_client_methods_received_topic_parse(&client, received_topic, &out_request)
+  );
+}
+
+static void test_az_iot_hub_client_methods_received_topic_parse_AZ_SPAN_NULL_received_topic_fail(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span received_topic = AZ_SPAN_NULL;
+
+  az_iot_hub_client_method_request out_request;
+
+  assert_precondition_checked(
+    az_iot_hub_client_methods_received_topic_parse(&client, received_topic, &out_request)
+  );
+}
+
+static void test_az_iot_hub_client_methods_received_topic_parse_NULL_out_request_fail(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span received_topic = AZ_SPAN_FROM_STR("$iothub/methods/POST/TestMethod/?$rid=1");
+
+  assert_precondition_checked(
+    az_iot_hub_client_methods_received_topic_parse(&client, received_topic, NULL)
+  );
+}
+
+#endif // NO_PRECONDITION_CHECKING
+
+static void test_az_iot_hub_client_methods_subscribe_topic_filter_get_succeed(void** state)
+{
+  (void)state;
+
+  uint8_t mqtt_sub_topic_buf[TEST_SPAN_BUFFER_SIZE];
+  memset(mqtt_sub_topic_buf, 0xFF, _az_COUNTOF(mqtt_sub_topic_buf));
```

Comment:
these can be updated using the changes John made with `az_span_for_test()`. Simplifies all of these.
[Link here](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/core/core/test/inc/az_test_span.h)

Reply:
Love it! Thank you @jspaith .

Reply:
Resolved.